docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,203 @@ The preceding C# code:
+var ep = new IPEndPoint(Socket.OSSupportsIPv6 ? IPAddress.IPv6Any : IPAddress.Any, 5000);
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+```
+
+### Start listening on server
+
+In `Socket` class there are two steps to start listening on a server, but in `TcpClient` calling `Start` function is enough. Consider the following TCP listener code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+listener.Start(10);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+socket.Bind(ep);
+try
+{
+    socket.Listen(10);
+}
+catch (SocketException)
+{
+    socket.Dispose();
+}
+```
+
+### Accepting a connection on server
+
+Basically, in `TcpListener` you can get two different type as return value on connection accept Those types are:
+
+- <xref:System.Net.Sockets.Socket>
+- <xref:System.Net.Sockets.TcpClient>
+
+You can get `TcpClient` from these functions: <xref:System.Net.Sockets.TcpListener.AcceptTcpClient> and <xref:System.Net.Sockets.TcpListener.AcceptTcpClientAsync>, and you can get `Socket` from these functions: <xref:System.Net.Sockets.TcpListener.AcceptSocket> and <xref:System.Net.Sockets.TcpListener.AcceptSocketAsync>.
+
+Consider the following TCP client code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+var acceptSocket = listener.AcceptSocket(); // var acceptSocket = listener.AcceptSocketAsync();
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+var acceptSocket = socket.Accept(); // var acceptSocket = await socket.AcceptAsync();
+```
+
+### Create a stream to send and receive data
+
+The difference in here is `TcpClient` class has a helper function for this if you want to create a `NetworkStream` which is called <xref:System.Net.Sockets.TcpClient.GetStream>, but in `Socket` class you should create it by yourself. Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient();
+NetworkStream stream = client.GetStream();
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+NetworkStream stream = new(socket, ownsSocket: true); // Please be aware, we're passing true as `ownsSocket` parameter to let stream object to own the socket. 
+```
+
+> [!TIP]
+> <xref:System.Net.Sockets.Socket> class has advantage if you don't want to use `Stream`. There are <xref:System.Net.Sockets.Socket.Send%2A>, <xref:System.Net.Sockets.Socket.SendAsync%2A>, <xref:System.Net.Sockets.Socket.Receive%2A> and <xref:System.Net.Sockets.Socket.ReceiveAsync%2A> overloads in `Socket` class.
```

Comment:
What advantage do you mean?

Reply:
You don't have to create `NetworkStream` if you're using `Socket`, basically you can use `Send`, `Receive` functions in `Socket`. I meant it.